### PR TITLE
[FEATURE] Implementation of Language model estimator

### PR DESCRIPTION
**Pull request by liuzh91**
## Description ##
Implementation of word language model estimator and large rnn language model estimator

### Essentials ###
- [ ] PR's title starts with a category (e.g. [BUGFIX], [MODEL], [TUTORIAL], [FEATURE], [DOC], etc)
- [ ] Changes are complete (i.e. I finished coding on this PR)
- [ ] All changes have test coverage
- [ ] Code is well-documented

### Changes ###
- [ ] Feature1, tests, (and when applicable, API doc)
- [ ] Feature2, tests, (and when applicable, API doc)

## Comments ##
- If this change is a backward incompatible change, why must this change be made.
- Interesting edge cases to note here

cc @dmlc/gluon-nlp-team

### CODE REVIEW

**Comment by mli** [comment]
Job PR-1155/1 is complete. 
Docs are uploaded to http://gluon-nlp-staging.s3-accelerate.dualstack.amazonaws.com/PR-1155/1/index.html

**Comment by mli** [comment]
Job PR-1155/2 is complete. 
Docs are uploaded to http://gluon-nlp-staging.s3-accelerate.dualstack.amazonaws.com/PR-1155/2/index.html

**Comment by mli** [comment]
Job PR-1155/4 is complete. 
Docs are uploaded to http://gluon-nlp-staging.s3-accelerate.dualstack.amazonaws.com/PR-1155/4/index.html

**Comment by mli** [comment]
Job PR-1155/5 is complete. 
Docs are uploaded to http://gluon-nlp-staging.s3-accelerate.dualstack.amazonaws.com/PR-1155/5/index.html

**Comment by mli** [comment]
Job PR-1155/6 is complete. 
Docs are uploaded to http://gluon-nlp-staging.s3-accelerate.dualstack.amazonaws.com/PR-1155/6/index.html

**Comment by mli** [comment]
Job PR-1155/7 is complete. 
Docs are uploaded to http://gluon-nlp-staging.s3-accelerate.dualstack.amazonaws.com/PR-1155/7/index.html

File: scripts/language_model/index.rst [hunk 1]
```diff
@@ -47,35 +47,35 @@ The dataset used for training the models is wikitext-2.
 
 For all the above model settings, we set Tied = True and NTASGD = True .
 
-[1] awd_lstm_lm_1150_wikitext-2 (Val PPL 68.71 Test PPL 65.62 )
+[1] awd_lstm_lm_1150_wikitext-2 (Val PPL 68.52 Test PPL 65.68 )
```
Comment (eric-haibin-lin) [inline scripts/language_model/index.rst]: While you're at it - would you mind removing the rows related to hyper-parameters in the table? i.e. Mode, Num_layers, Embed size, Hidden size, Dropout, Dropout_h, Dropout_i, Dropout_e, Weight_drop. After the removal the table will be simpler. Also could you move the command to https://github.com/dmlc/web-data/tree/master/gluonnlp/logs/language_model and reference the links in the table? Currently, the commands take a lot of space. We shall simply the table for cache_lm and large word lm, too.

File: scripts/language_model/index.rst [hunk 2]
```diff
@@ -181,8 +181,8 @@ The dataset used for training the models is Google's 1 billion words dataset.
 
 .. code-block:: console
 
-   $ python large_word_language_model.py --gpus 0,1,2,3 --clip=10
-   $ python large_word_language_model.py --gpus 4 --eval-only --batch-size=1
+   $ python large_word_language_model_estimator.py --gpus 0,1,2,3 --clip=10
+   $ python large_word_language_model_estimator.py --gpus 4 --eval-only --batch-size=1
```
Comment (eric-haibin-lin) [inline scripts/language_model/index.rst]: No PPL change for large_word_language_model?

Comment (liuzh91) [reply scripts/language_model/index.rst]: I am still training on large_word_language_model. It takes approximately 6~7 days to finish the whole training. Currently I get a test ppl of 43.98 with the latest model checkpoint which is comparable to the baseline model. I will update it after the training is completed.

**Comment by mli** [comment]
Job PR-1155/8 is complete. 
Docs are uploaded to http://gluon-nlp-staging.s3-accelerate.dualstack.amazonaws.com/PR-1155/8/index.html

File: scripts/language_model/index.rst (hunk 1 above)
Comment (liuzh91) [reply scripts/language_model/index.rst]: Doc is updated. I have submitted a new PR https://github.com/dmlc/web-data/pull/232 to move the commands.

**Comment by mli** [comment]
Job PR-1155/9 is complete. 
Docs are uploaded to http://gluon-nlp-staging.s3-accelerate.dualstack.amazonaws.com/PR-1155/9/index.html

**Comment by mli** [comment]
Job PR-1155/10 is complete. 
Docs are uploaded to http://gluon-nlp-staging.s3-accelerate.dualstack.amazonaws.com/PR-1155/10/index.html
